docs/models/advanced/imodel.md

```diff
@@ -0,0 +1,159 @@
+`IModel`
+========
+
+`IModel` is a type that represents a model in its raw state. This is how a model is represented right after it has been loaded. Usually this directly represents the source of the model (e.g. an object deserialized from JSON, or an OBJ container). It also defines some important methods.
+
+If an `IModel` also implements certain subinterfaces of `IModel`, it can also process some extra information. External entities using the model can do this processing through the static helpers in `ModelProcessingHelper`.
+
+At this high level, a model has no concept of items, blocks, or anything of that sort; it purely represents a shape.
+
+!!! important
+    `IModel` is immutable. Methods such as `IModelCustomData::process` that alter the model should never modify the `IModel`, as they should construct new `IModel`s instead.
+
+### `getDependencies`
+
+This is a collection of the `ResourceLocation`s of all the models this model depends on. These models are guaranteed to be loaded before this one is baked. For example, a model deserialized from a blockstate JSON will depend on the models defined within. Only models that are directly mapped to a block/item are loaded normally; to ensure loading of other models, they must be declared as dependencies of another. Cyclic dependencies will cause a `LoaderException` to be thrown.
+
+### `getTextures`
+
+This is a collection of the `ResourceLocation`s of all the textures this model depends on. These textures are guaranteed to be loaded before this model is baked. For example, a vanilla JSON model depends on all the textures defined within.
+
+### `bake`
+
+This is the main method of `IModel`. It takes an `IModelState`, a `VertexFormat`, and a function `ResourceLocation` → `TextureAtlasSprite`, to return an `IBakedModel`. `IBakedModel` is lower-level than `IModel`, and it is what interacts with blocks and items. The function RL → TAS is used to get textures from RLs (i.e. the RLs of textures are passed to this function and the returned TAS contains the texture).
+
+`IAnimatedModel`
+----------------
+
+`IAnimatedModel` is part of the currently WIP animation API. Right now it does nothing.
```

Comment:
I saw no references to it when I searched, but oh well. Will a "TODO: explain" do, or should I just replace the entire section with a "TODO: `IAnimatedModel`".